core/src/main/java/org/apache/brooklyn/core/sensor/password/CreatePasswordSensor.java

```diff
@@ -33,23 +34,21 @@
 
     public static final ConfigKey<String> ACCEPTABLE_CHARS = ConfigKeys.newStringConfigKey("password.chars", "The characters allowed in password");
 
-    private Integer passwordLength;
-    private String acceptableChars;
-
     public CreatePasswordSensor(Map<String, String> params) {
         this(ConfigBag.newInstance(params));
     }
 
     public CreatePasswordSensor(ConfigBag params) {
         super(params);
-        passwordLength = params.get(PASSWORD_LENGTH);
-        acceptableChars = params.get(ACCEPTABLE_CHARS);
     }
 
     @Override
     public void apply(EntityLocal entity) {
         super.apply(entity);
 
+        Integer passwordLength = EntityInitializers.resolve(params, PASSWORD_LENGTH);
+        String acceptableChars = EntityInitializers.resolve(params, ACCEPTABLE_CHARS);
```

Comment:
`apply` is called before managing any of the entities in the spec hierarchy (i.e. a camp plan) so this can't be resolved at this point. It would just block app creation. Could you add some tests to prove me wrong :). 
`StaticSensor` does the right thing by pushing all of it in tasks.